### PR TITLE
SAM0 cpu: persist RTC GP registers across reset (see #16756)

### PR DESCRIPTION
**Pull request by ant9000**
As discussed in #16756 - a simple solution for persisting RTC general purpose registers across RTC reset.

### CODE REVIEW

File: cpu/sam0_common/periph/rtc_rtt.c [hunk 1]
```diff
@@ -135,13 +135,25 @@ static inline void _rtc_set_enabled(bool on)
 
 static inline void _rtt_reset(void)
 {
+#if CONFIG_SAM0_PERIPH_RTT_RTC_PERSIST_GPR
+#ifdef RTC_GPR_NUM
+    uint32_t persist[RTC_GPR_NUM];
+    int i;
+    for (i = 0; i < RTC_GPR_NUM; i++) persist[i] = RTC->MODE0.GP[i].reg;
```
Comment (benpicco) [inline cpu/sam0_common/periph/rtc_rtt.c:142]:
```suggestion
    for (unsigned i = 0; i < RTC_GPR_NUM; i++) {
        persist[i] = RTC->MODE0.GP[i].reg;
    }
```

No need to squish that all on one line :wink:

File: cpu/sam0_common/periph/rtc_rtt.c [hunk 2]
```diff
@@ -135,13 +135,25 @@ static inline void _rtc_set_enabled(bool on)
 
 static inline void _rtt_reset(void)
 {
+#if CONFIG_SAM0_PERIPH_RTT_RTC_PERSIST_GPR
+#ifdef RTC_GPR_NUM
```
Comment (benpicco) [inline cpu/sam0_common/periph/rtc_rtt.c:139]:
```suggestion
#if IS_ACTIVE(CONFIG_SAM0_PERIPH_RTT_RTC_PERSIST_GPR) && defined(RTC_GPR_NUM)
```

**Comment by benpicco** [comment]
Which reminds me: We should probably add a proper API for those general purpose RTC registers since there are many RTCs that have those.

But this is a good start.

File: cpu/sam0_common/periph/rtc_rtt.c [hunk 3]
```diff
@@ -135,13 +135,25 @@ static inline void _rtc_set_enabled(bool on)
...
 #ifdef RTC_MODE0_CTRL_SWRST
     RTC->MODE0.CTRL.reg = RTC_MODE0_CTRL_SWRST;
     while (RTC->MODE0.CTRL.bit.SWRST) {}
 #else
     RTC->MODE0.CTRLA.reg = RTC_MODE2_CTRLA_SWRST;
     while (RTC->MODE0.CTRLA.bit.SWRST) {}
 #endif
+#if CONFIG_SAM0_PERIPH_RTT_RTC_PERSIST_GPR
+#ifdef RTC_GPR_NUM
+    for (i = 0; i < RTC_GPR_NUM; i++) RTC->MODE0.GP[i].reg = persist[i];
```
Comment (benpicco) [inline cpu/sam0_common/periph/rtc_rtt.c:154]: don't we have to wait for SYNCBUSY before we can write to the next `RTC->MODE0.GP` register?

**Comment by ant9000** [comment]
This PR is superseded by #16758 - let's close it.

**Pull request closed without merging**